### DISCUSSION TITLE
Timezone plugin?  Could not find time zone with id America/New_York

### DISCUSSION THREAD

**Discussion by s3YwCf2ZbfJG4SHAfjQMAjtsf** [quartznet/quartznet#1958] [Q&A]
If I attempt TimeZoneInfo.FindSystemTimeZoneById("America/New_York") locally, it returns US Eastern.   The server doesn't seem to be able to find it.  Although, my local machine is Windows 11, my server is Windows 2019.

I am about to implement the TimeZone plugin as suggested by the error and see what happens.   Not sure if I am encountering a real Windows problem?

2023-03-19 14:41:25,993 [26] ERROR Quartz.Impl.AdoJobStore.JobStoreTX - Error retrieving the misfired trigger: 'HealthCore.DataArcJob_212_158'
Quartz.JobPersistenceException: Couldn't retrieve trigger: **Could not find time zone with id America/New_York, consider using Quartz.Plugins.TimeZoneConverter for resolving more time zones ids**
 ---> System.TimeZoneNotFoundException: Could not find time zone with id America/New_York, consider using Quartz.Plugins.TimeZoneConverter for resolving more time zones ids
 ---> System.TimeZoneNotFoundException: **The time zone ID 'America/New_York' was not found on the local computer.**
   at System.TimeZoneInfo.FindSystemTimeZoneById(String id)
   at Quartz.Util.TimeZoneUtil.FindTimeZoneById(String id)

**Comment by jafin** [comment]
Slightly related, similar issues on Azure Function (https://github.com/Azure/azure-functions-host/issues/7909).  So it seems available timezones are not consistent across platforms/OS.

**Comment by lahma** [comment]
NET 6 should also offer better compatibility.

**Comment by s3YwCf2ZbfJG4SHAfjQMAjtsf** [comment]
The app is .Net 6, but some machines report the error.

**Comment by s3YwCf2ZbfJG4SHAfjQMAjtsf** [comment]
The plugin cleared the errors but caused the West Coast jobs to be treated as Eastern.   Still collecting data.   All jobs triggered this time.

**Comment by s3YwCf2ZbfJG4SHAfjQMAjtsf** [comment]
A little hack:  (tzInfo is the TimezoneInfo based on the IANA with the incorrect Id)

           var tzCollection = TimeZoneInfo.GetSystemTimeZones();                
           
           var matchingTz = tzCollection.FirstOrDefault(tz => tz.StandardName == tzInfo.StandardName);
                if (matchingTz != null)
                {
                    item.WindowsTimeZone = matchingTz.Id;
                }
                else
                {
                    item.WindowsTimeZone = tzInfo.Id;
                }
